From 1d8e47170bf62c536075037b80dc31848adbf4ad Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Sun, 20 Nov 2022 20:20:43 +0800
Subject: [PATCH 20/56] Revert "drm/nouveau/nouveau_bo: fix potential memory
 leak in nouveau_bo_alloc()"

This reverts commit 3339a51bcd89fce33011abf226b17c1c522b5e24.
---
 drivers/gpu/drm/nouveau/nouveau_bo.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/nouveau/nouveau_bo.c b/drivers/gpu/drm/nouveau/nouveau_bo.c
index b57dcad8865f..b4946b595d86 100644
--- a/drivers/gpu/drm/nouveau/nouveau_bo.c
+++ b/drivers/gpu/drm/nouveau/nouveau_bo.c
@@ -279,10 +279,8 @@ nouveau_bo_alloc(struct nouveau_cli *cli, u64 *size, int *align, u32 domain,
 			break;
 	}
 
-	if (WARN_ON(pi < 0)) {
-		kfree(nvbo);
+	if (WARN_ON(pi < 0))
 		return ERR_PTR(-EINVAL);
-	}
 
 	/* Disable compression if suitable settings couldn't be found. */
 	if (nvbo->comp && !vmm->page[pi].comp) {
-- 
2.25.1

